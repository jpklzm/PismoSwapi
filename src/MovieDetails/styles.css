.title {
  color: white;
  margin: auto;
  width: 50%;
  padding: 10px;
  text-align: center;
  font-size: 50px;
}

h2 {
  color: yellow;
  margin: auto;
  width: 50%;
  padding: 10px;
  text-align: center;
  font-size: 35px;
}

.backbutton {
  background-color: rgb(100, 100, 100, 0.5);
  padding: 15px 30px;
  border-radius: 10px;
  margin: 5px;
  color: yellow;
  cursor: pointer;
  position: absolute;
  right: 0;
  top: 0;
}

.backbutton:hover {
  background-color: rgb(100, 100, 100, 1);
}
