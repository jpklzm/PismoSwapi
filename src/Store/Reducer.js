const Reducer = (state, action) => {
  switch (action.type) {
    case "SET_CURRENT_MOVIE":
      return {
        ...state,
        movie: action.payload,
      };
    case "SET_CURRENT_MOVIE_LIST":
      return {
        ...state,
        movieList: action.payload,
      };
    default:
      return state;
  }
};

export default Reducer;
