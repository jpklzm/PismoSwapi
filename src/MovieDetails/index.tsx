import React, { FunctionComponent, useContext, useCallback } from "react";
import "./styles.css";
import { Context } from "../Store/Store";
import { useHistory } from "react-router-dom";

const MovieDetails = () => {
  //@ts-ignore
  const [state, dispatch] = useContext(Context);
  const history = useHistory();
  const handleOnClick = useCallback(() => history.push("/"), [history]);

  return (
    <div>
      <h1 className="title">{state.movie.title}</h1>
      <h2>{`Episode ${state.movie.episode_id}`}</h2>
      <h2>{`Director: ${state.movie.director}`}</h2>
      <h2>{`Producer: ${state.movie.producer}`}</h2>
      <button className="backbutton" onClick={handleOnClick}>
        Back
      </button>
    </div>
  );
};

export default MovieDetails;
