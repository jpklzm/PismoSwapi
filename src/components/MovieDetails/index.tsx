import React, { useContext, useCallback } from "react";
import "./styles.css";
import { Context } from "../Store/Store";
import { useHistory } from "react-router-dom";

const MovieDetails = () => {
  //@ts-ignore
  const [state, dispatch] = useContext(Context);
  const history = useHistory();
  const handleOnClick = useCallback(() => history.push("/"), [history]);

  const getWeekday = (day: number) => {
    if (day === 0) return "Domingo";
    if (day === 1) return "Segunda-feira";
    if (day === 2) return "Terça-feira";
    if (day === 3) return "Quarta-feira";
    if (day === 4) return "Quinta-feira";
    if (day === 5) return "Sexta-feira";
    if (day === 6) return "Sábado";
  };

  const getMonthName = (month: number) => {
    if (month === 0) return "Janeiro";
    if (month === 1) return "Fevereiro";
    if (month === 2) return "Março";
    if (month === 3) return "Abril";
    if (month === 4) return "Maio";
    if (month === 5) return "Junho";
    if (month === 6) return "Julho";
    if (month === 7) return "Agosto";
    if (month === 8) return "Setembro";
    if (month === 9) return "Outubro";
    if (month === 10) return "Novembro";
    if (month === 11) return "Dezembro";
  };

  const formatedDate = `${getWeekday(
    new Date(state.movie.edited).getDay()
  )}, ${new Date(state.movie.edited).getDate()} de ${getMonthName(
    new Date(state.movie.edited).getMonth()
  )} de ${new Date(state.movie.edited).getFullYear()}`;

  return (
    <div>
      <h1 className="title">{state.movie.title}</h1>
      <h2>{`Edited: ${formatedDate}`}</h2>
      <h2>{`Episode ${state.movie.episode_id}`}</h2>
      <h2>{`Director: ${state.movie.director}`}</h2>
      <h2>{`Producer: ${state.movie.producer}`}</h2>
      <button className="backbutton" onClick={handleOnClick}>
        Back
      </button>
    </div>
  );
};

export default MovieDetails;
