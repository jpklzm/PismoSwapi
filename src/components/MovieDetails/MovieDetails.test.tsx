import { mount } from "enzyme";
import MovieDetails from "./index";
import { Context } from "../Store/Store";

const contextValues = {
  movie: { title: "filme", producer: "produtor", director: "diretor" },
};

describe("MovieDetails", () => {
  it("renders correctly", () => {
    const wrapper = mount(
      <Context.Provider value={[contextValues, ""]}>
        <MovieDetails />
      </Context.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
