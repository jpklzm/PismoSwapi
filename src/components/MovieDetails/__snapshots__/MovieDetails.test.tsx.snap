// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails renders correctly 1`] = `
<MovieDetails>
  <div>
    <h1
      className="title"
    >
      filme
    </h1>
    <h2>
      Episode undefined
    </h2>
    <h2>
      Director: diretor
    </h2>
    <h2>
      Producer: produtor
    </h2>
    <button
      className="backbutton"
      onClick={[Function]}
    >
      Back
    </button>
  </div>
</MovieDetails>
`;
